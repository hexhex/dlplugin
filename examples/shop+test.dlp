% -*- prolog -*-

#namespace("shop", "http://www.kr.tuwien.ac.at/staff/roman/shop#")
#namespace("shop", "http://www.test.com/test#")

plusProvides("provides","foo","moo").
plusPart("Part","cpu1").
minusPart("Part","cpu2").
minusShop("Shop","not_a_shop").

s0(X) :- &dlC["shop.owl",a,b,c,d,"Shop"](X).
s2(Y) :- &dlC["shop.owl",a,b,c,d,"Shop"](Y).

s1(X) :- &dlC["shop.owl",plusPart,minusPart,plusProvides,foo,"Shop"](X).

p0(X,Y) :- &dlR["shop.owl",plusPart,minusPart,plusProvides,foo,"provides"](X,Y), s0(X).

p1(A,B) :- &dlR["shop.owl",plusPart,minusPart,plusProvides,foo,"provides"](A,B), not s1(B).

part(X) :- &dlC["shop.owl",plusPart,minusPart,plusProvides,foo,"Part"](X).

nshop1(X) :- &dlC["shop.owl",a,b,c,d,"-Shop"](X).
nshop2(X) :- &dlC["shop.owl",plusPart,minusShop,plusProvides,foo,"-Shop"](X).


tester1(X) :- &dlC["test.owl",a,b,c,d,"Test"](X).
tester2(X) :- &dlC["test.owl","a","b","c","d","Test"](X).
