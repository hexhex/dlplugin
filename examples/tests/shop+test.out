{tester2("test:test1"), tester2("test:test2"), tester2("test2:test1"), tester1("test:test1"), tester1("test:test2"), tester1("test2:test1"), nshop2("shop:not_a_shop"), part("shop:cpu1"), part("shop:moo"), part("shop:fan"), part("shop:wlan"), part("shop:graphiccard"), part("shop:cdrom"), part("shop:harddisk"), part("shop:dvdrom"), part("shop:memory"), part("shop:soundcard"), part("shop:cpu"), part("shop:case"), p1("shop:foo","shop:moo"), p1("shop:s1","shop:cpu"), p1("shop:s1","shop:case"), p1("shop:s2","shop:soundcard"), p1("shop:s3","shop:dvdrom"), p1("shop:s3","shop:memory"), p1("shop:s3","shop:cpu"), p1("shop:s4","shop:cdrom"), p1("shop:s4","shop:harddisk"), p1("shop:s8","shop:cdrom"), p1("shop:s8","shop:dvdrom"), p1("shop:s8","shop:memory"), p1("shop:s8","shop:soundcard"), p1("shop:s5","shop:harddisk"), p1("shop:s5","shop:cpu"), p1("shop:s6","shop:graphiccard"), p1("shop:s6","shop:case"), p1("shop:s7","shop:wlan"), p1("shop:s7","shop:harddisk"), p1("shop:s9","shop:harddisk"), p1("shop:s9","shop:case"), p1("shop:s10","shop:fan"), p0("shop:s1","shop:cpu"), p0("shop:s1","shop:case"), p0("shop:s2","shop:soundcard"), p0("shop:s3","shop:dvdrom"), p0("shop:s3","shop:memory"), p0("shop:s3","shop:cpu"), p0("shop:s4","shop:cdrom"), p0("shop:s4","shop:harddisk"), p0("shop:s8","shop:cdrom"), p0("shop:s8","shop:dvdrom"), p0("shop:s8","shop:memory"), p0("shop:s8","shop:soundcard"), p0("shop:s5","shop:harddisk"), p0("shop:s5","shop:cpu"), p0("shop:s6","shop:graphiccard"), p0("shop:s6","shop:case"), p0("shop:s7","shop:wlan"), p0("shop:s7","shop:harddisk"), p0("shop:s9","shop:harddisk"), p0("shop:s9","shop:case"), p0("shop:s10","shop:fan"), s1("shop:foo"), s1("shop:s1"), s1("shop:s2"), s1("shop:s3"), s1("shop:s4"), s1("shop:s8"), s1("shop:s5"), s1("shop:s6"), s1("shop:s7"), s1("shop:s9"), s1("shop:s10"), s2("shop:s1"), s2("shop:s2"), s2("shop:s3"), s2("shop:s4"), s2("shop:s8"), s2("shop:s5"), s2("shop:s6"), s2("shop:s7"), s2("shop:s9"), s2("shop:s10"), s0("shop:s1"), s0("shop:s2"), s0("shop:s3"), s0("shop:s4"), s0("shop:s8"), s0("shop:s5"), s0("shop:s6"), s0("shop:s7"), s0("shop:s9"), s0("shop:s10"), minusShop("Shop","not_a_shop"), minusPart("Part","cpu2"), plusPart("Part","cpu1"), plusProvides("provides","foo","moo")}
